### DISCUSSION TITLE
EDAX and MTEX orientation maps comparison

### DISCUSSION THREAD

**Discussion by Antonio-Silveira** [mtex-toolbox/mtex#1400] [Ask Anything]
Hello everyone,

I plot the ipf oriented at the directions 100, 010, and 001 from the exact same data in Mtex and at the local EDAX software in the lab. I am comparing because I wanted to make sure that I am not messing anything up in my Mtex scrip.

So, the actual map shows the exact same pattern, however, the color maps for the 100 and 010 are inverted in Mtex and I don't understand why. If I am making the correct import, I should see the exact same results, correct?

I modify the x, and y directions and also the lattice(a,b,c) directions to obtain the same color map pattern, but by doing this, the actual image is rotated. The EDAX software is using the "Setting 2" and therefore.

in Mtex, I am using the following convention:
```matlab
% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('yAxisDirection','north');
setMTEXpref('zAxisDirection','intoPlane');

setMTEXpref('aAxisDirection','north');
setMTEXpref('bAxisDirection','east');
```
and during the import, I am converting the Euler (A1,A2,A3) directions to match the sample directions (X,Y,Z). 
```matlab
ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertEuler2SpatialReferenceFrame','setting 2');
 ```
**By looking to the results from Mtex, I would say that many 111 planes are aligned with the X direction(100) while by looking at the EDAX figures, I would say that the 111 planes are actually aligned with the Y direction (010).** 

![mtex_edax](https://user-images.githubusercontent.com/103099688/163005057-6b3371c4-c611-4e95-b4fb-b990835fc95a.png)

Here is a photo that I took from the EDAX reference at my institute:
![screenshot](https://user-images.githubusercontent.com/103099688/163003673-5b9242c7-b81f-483d-9623-2970eb2bc2f2.jpg)

I know that this is a very basic topic and it is very well discussed in the documentation, here in the forum, and also during the Mtex workshop, but I still don't get it.

I would like to thank anyone in advance for pointing me to the right direction (no pun intended)

Best regards,
Antonio

**Comment by Tijmenvermeij** [comment]
Hi,

I would argue that your Mtex X IPF map is the same as your Edax Y IPF map. Are you sure you're plotting the correct IPF both for Mtex and Edax?

**Reply by Antonio-Silveira** [reply]
Hi,
That's right! My Mtext X IPF maps is the Edax Y IPF. 

I tried to change the plot convention in Mtex to reach the exact same result, but I always end it up rotating the figure, and not changing the direction in which the color map is referred.  I don't know how to change this.

Here is how I am plotting these maps in Mtex:
```matlab
%% OM 100 
cs = CS{2};
ck = ipfHSVKey(cs)
ck.inversePoleFigureDirection = xvector;
figure(5)
plot(ebsdS('Martensite'),ck.orientation2color(ebsdS('Martensite').orientations))
hold on
plot(grainsS.boundary,'lineWidth',1,'micronbar','off')
hold off
mtexTitle(['ref.dir.: ' char(xvector)])
%% OM 010
cs = CS{2};
ck = ipfHSVKey(cs)
ck.inversePoleFigureDirection = yvector;
figure(6)
plot(ebsdS('Martensite'),ck.orientation2color(ebsdS('Martensite').orientations))
hold on
plot(grainsS.boundary,'lineWidth',1,'micronbar','off')
hold off
mtexTitle(['ref.dir.: ' char(yvector)])
```

**Reply by Tijmenvermeij** [reply]
Ok, and you're aware that in Edax you need to choose A2 IPF to get the X-direction IPF?
Also, your mtex plotting preference do not form a right handed coordinate system; not sure if this is an issue...

**Reply by kilir** [reply]
The observation is unrelated to the plotting convention and exactly the difference between A123 and xyz reference frames.

'setting 2' corresponds to `rotation.byAxisAngle(xvector-yvector,180*degree)`, so in the case of specimen directions, this should be (1,0,0)_Edax -> (0,-1,0)_mtex. I'd say you won't see the difference between (0,-1,0) and (0,1,0) in your material.

Cheers,
Rüdiger

**Reply by Tijmenvermeij** [reply]
Of course, one should not compare Mtex X with Edax A1...

**Comment by Antonio-Silveira** [comment]
Hi everyone, thanks so much for your comments!

I managed to obtain the exact same result from Edax. Thanks for pointing out the coordinate system, it was wrong.
But to get the same result I aligned A1 to y and A2 to x.

For this I used 
```matlab 
setMTEXpref('xAxisDirection','north');
setMTEXpref('zAxisDirection','OutOfPlane');

ebsd = EBSD.load(fname,CS,'interface','ang',...
  'convertSpatial2EulerReferenceFrame','setting 2');
```
This will change for every user of course, but the key thing for me was moving the spatial coordinates and not the Euler. i.e. using:
```matlab 
'convertSpatial2EulerReferenceFrame'
 ```
instead of  
```matlab 
'convertEuler2SpatialReferenceFrame' 
```
here is some Pole figures from Edax and Mtex now.
![grafik](https://user-images.githubusercontent.com/103099688/164235243-95fba049-b88f-4dd8-b7f7-978da7f2196d.png)
